text-feature-translation/src/main/java/net/kyori/adventure/text/feature/translation/TranslationRegistryImpl.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of adventure, licensed under the MIT License.
+ *
+ * Copyright (c) 2017-2020 KyoriPowered
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package net.kyori.adventure.text.feature.translation;
+
+import net.kyori.adventure.key.DuplicateKeyException;
+import net.kyori.adventure.key.Key;
+import net.kyori.examination.Examinable;
+import net.kyori.examination.ExaminableProperty;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.text.MessageFormat;
+import java.util.Comparator;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.stream.Stream;
+
+/* package */ final class TranslationRegistryImpl extends TranslatableComponentRenderer implements TranslationRegistry {
+  /* package */ static final TranslationRegistry INSTANCE = new TranslationRegistryImpl();
+
+  private static final class Translation implements Key, Examinable {
+    private final String owner;
+    private final String key;
+    private final Map<Locale, MessageFormat> formats;
+
+    private Translation(final @NonNull Key key) {
+      this.owner = key.namespace();
+      this.key = key.value();
+      this.formats = new ConcurrentSkipListMap<>(Comparator.comparing(Locale::toLanguageTag));
```

Comment:
Does this have to be a `ConcurrentSkipListMap`? This could just as easily been a `ConcurrentHashMap` (I assume you meant to make this class thread-safe).